### PR TITLE
Asintel 1770 marketo data retention

### PR DESCRIPTION
**Pull request by sarmaevanidfe**

### CODE REVIEW

**Review by uhimabindu — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.DATA.MANAGEMENT/Tables/AsDataPL_Marketo_Leads.sql [hunk 1]
```diff
@@ -6,8 +6,12 @@
 ,EmailAddress nvarchar(255)
 ,LeadCreatedAt datetime2
 ,LeadUpdatedAt datetime2
+,EmployerHashedId  nvarchar(100)
+,ProviderId  bigint NULL
```
[the remark this answers is not in the document]
Comment (sarmaevanidfe) [reply src/SFA.DAS.DATA.MANAGEMENT/Tables/AsDataPL_Marketo_Leads.sql:10]: Now managed to load EmployerHashedID and Provider to Marketo lead table

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/ImportMarketoBulkExtractTablesToPL.sql [hunk 2]
```diff
@@ -44,9 +44,18 @@ BEGIN TRANSACTION
 /* Delta Code */
 
 /* Delta Update MarketoLeads */
-
+;with baseMarketoLeadsData as
+(
+		Select distinct MLData.LeadId,MLData.[FirstName],MLData.[LastName],MLData.[EmailAddress],MLData.[CreatedAt],MLData.[UpdatedAt],EmpData.EmployerHashedId,ProviderData.Ukprn As ProviderID
+		from stg.MarketoLeads  MLData   LEFT JOIN
```
Comment (uhimabindu) [inline src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/ImportMarketoBulkExtractTablesToPL.sql:50]: As Marketo Imports are Delta, If we joining on Staging Table to get Employer HashedId and Provider Id then we will have them only for latest but will miss on the data that is already in PL

**Review by uhimabindu — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/ImportMarketoBulkExtractTablesToPL.sql (hunk 2 above)
Comment (sarmaevanidfe) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/ImportMarketoBulkExtractTablesToPL.sql:79]: As discussed, add the update statement for existing data updates to marketo leads table

Comment (sarmaevanidfe) [reply src/SFA.DAS.DATA.MANAGEMENT/StoredProcedure/ImportMarketoBulkExtractTablesToPL.sql:68]: Amend the update statement for EmployerHashedID and ProviderID

**Review by uhimabindu — APPROVED** [review]

**Pull request merged**